test/config/utility.cc

```diff
@@ -491,6 +492,33 @@ void ConfigHelper::addConfigModifier(HttpModifierFunction function) {
   });
 }
 
+void blockFormat(YAML::Node node) {
+  node.SetStyle(YAML::EmitterStyle::Block);
+
+  if (node.Type() == YAML::NodeType::Sequence) {
+    for (auto it : node) {
+      blockFormat(it);
+    }
+  }
+  if (node.Type() == YAML::NodeType::Map) {
+    for (auto it : node) {
+      blockFormat(it.second);
+    }
+  }
+}
+
+const std::string ConfigHelper::bootstrapAsYaml() {
+  std::string json;
+  const auto status = Protobuf::util::MessageToJsonString(bootstrap_, &json);
+  RELEASE_ASSERT(status.ok(), "Failed to convert bootstrap to Json");
+  auto node = YAML::Load(json);
+  blockFormat(node);
+
+  YAML::Emitter out;
+  out << node;
+  return out.c_str();
```

Comment:
Can we package up the proto -> YAML utility and put it in https://github.com/envoyproxy/envoy/blob/master/source/common/protobuf/utility.h#L151? Maybe also add a few simple tests.

This is a neat utility, there are other places we might want it, e.g. when outputing trace files in YAML file instead of JSON.

Reply:
Specifically, https://github.com/envoyproxy/envoy/blob/master/source/common/protobuf/utility.h#L290 is the companion I had in mind.

Reply:
Ehhh, the recursion for block format seemed a bit hacky and the formatting isn't quite ideal, but I suppose an imperfect util is better than none?

Sample output below.  The {} on a line by itself bugs me and could be fixed with some massaging but that's more than I wanted to invest for my test util today.  Just noting for the record and I'll capture in test as well.  Overall it's still way better than running the test with trace on, and doing 2 manual conversions :-P

static_resources:
  listeners:
    - name: http
      address:
        socket_address:
          address: 127.0.0.1
          port_value: 0
      filter_chains:
        - filters:
            - name: envoy.http_connection_manager
              config:
                route_config:
                  virtual_hosts:
                    - routes:
                        - route:
                            cluster: cluster_0
                          match:
                            prefix: /
                      domains:
                        - "*"
                      name: integration
                  name: route_config_0
                codec_type: HTTP2
                http_filters:
                  - name: envoy.router
                stat_prefix: config_test
  clusters:
    - name: cluster_0
      connect_timeout: 5s
      hosts:
        - socket_address:
            address: 127.0.0.1
            port_value: 45549
      http2_protocol_options:
        {}
admin:
  access_log_path: /dev/null
  address:
    socket_address:
      address: 127.0.0.1
      port_value: 0

Reply:
Huge +1 from me on this, I actually wanted this functionality in a PR I was just working on.